### PR TITLE
Release candidate

### PR DESCRIPTION
**Pull request by raphinesse**
I tested this in two of our repos and it worked as intended. Only gotcha is that linked installations didn't work. I had to install from packages generated with `npm pack` or ESLint's module resolution would choke. But I don't think it will be an issue when installed from the registry or the repo.

**IMHO this could be released as is.**

## Questions
- Should we also alias `cordova/node-tests` to `cordova/tests` be a better counterpart name to `cordova`?
- If this should be a scoped package, we would have to adjust a few things.
  - Package name would be `@cordova/eslint-config`
  - `extends: cordova` would become `extends: @cordova`
  - `extends: cordova/browser` would become `extends: @cordova/eslint-config/browser` which is a bit ugly but no deal breaker
- Should we already prep the repo structure for multiple packages (i.e. move everything into a dir `eslint-config-cordova`)?
  - I'd say we can still do this if/when we actually want to add a second package
- Add `engines` to `package.json`?

## Remarks
- In my repo there are **some more branches**, _including one for v2_ and some for config variations that exist throughout Cordova's repos.
<details>
  <summary>List of ESLint configs used in our repos</summary>

# Linting rules in use across Cordova repos

### Unchanged
- cordova-android
- cordova-browser
- cordova-cli
- cordova-coho
- cordova-create
- cordova-docs
- cordova-electron
- cordova-fetch
- cordova-ios
- cordova-plugin-battery-status
- cordova-plugin-camera
- cordova-plugin-device
- cordova-plugin-dialogs
- cordova-plugin-file
- cordova-plugin-geolocation
- cordova-plugin-inappbrowser
- cordova-plugin-media-capture
- cordova-plugin-network-information
- cordova-plugin-vibration
- cordova-plugin-wkwebview-engine
- cordova-plugman
- cordova-serve
- cordova-windows
- cordova-osx

### Node + linebreak-style + promise/catch-or-return
- cordova-lib

### Node stricter
- cordova-js

### Node stricter + no-unused-args-after-used
- cordova-common

### Unchanged
- cordova-android/cordova-js-src
- cordova-browser/cordova-js-src
- cordova-js/test/test-platform-modules

### Browser + ES5 only + custom globals
- cordova-js/src

## Browser Tests
- cordova-js/test

### Unchanged
- cordova-browser/spec
- cordova-cli/spec/.eslintrc.yaml
- cordova-create/spec
- cordova-electron/tests
- cordova-ios/tests/spec
- cordova-plugin-inappbrowser/tests
- cordova-plugman/spec
- cordova-windows/spec
- cordova-osx/tests/spec

### Node Tests + expectAsync fix
- cordova-common/spec

### Node Tests - prefer-promise-reject-errors
- cordova-android/spec
- cordova-fetch/spec
- cordova-lib/integration-tests
- cordova-lib/spec
</details>

### CODE REVIEW

**Comment by erisu** [comment]
> Should we also alias cordova/node-test to cordova/test be a better counterpart name to cordova?

Do you see node and browser being different? 

I think `cordova/test` is cleaner but do we need to be descriptive enough if there are differences between node and browser test.

> If this should be a scoped package, we would have to adjust a few things.
> Package name would be @cordova/eslint-config
> extends: cordova would become extends: @cordova
> extends: cordova/browser would become extends: @cordova/eslint-config/browser which is a bit ugly but no deal breaker

I am not against starting to use scope packages.

> Should we already prep the repo structure for multiple packages (i.e. move everything into a dir eslint-config-cordova)?
> I'd say we can still do this if/when we actually want to add a second package

I was debating on if we would actually be releasing two packages from the beginning and if so, we should start doing multiple package structure.

Example: In my current repo master, all of the needed dependencies are set in the `package.json` as `dependencies`. 

Most or all of the `eslint-config-*` and `eslint-plugin-*` out there sets them as `peerDependencies`. This is to not enforce the dependencies on the users project.

For Cordova's sake, we want to enforce the dependencies for consistency across repos. I think we wanted to manage both dependencies and configs in one area.

But, if we want to share our config/plugin to Cordova plugin developers, do we not want to make the package in the same manner as others packages? not enforce the dependencies.

Basically what this might mean is creating two package, one being our core plugin called `cordova-eslint` which has the dependencies set in `package.json` as `dependencies` and is installed on each repo, and also the core package would also contain our new `eslint-config-cordova` or plugin.

> Add engines to package.json?

Hm maybe we could as because some of the other plugins and configs do so as well. With an engine set maybe people will think of looking into the other dependencies engine requirement before upgrading.

**Comment by raphinesse** [comment]
> Do you see node and browser being different?

Well they _already are different now_ because of the different `env`. Plus, there _is_ potential for more divergence (e.g. [my `es5-only` branch](https://github.com/raphinesse/eslint-config-cordova/commit/e36067ebe5c94a509f707bd6a812bf644d333b35)). I would definitely roll them out as different (sub-)configs, as that allows us to roll out changes more easily across repos.

> I think `cordova/test` is cleaner but do we need to be descriptive enough if there are differences between node and browser test.

As I said, `cordova/tests` would also only be an _alias_ to `cordova/node-tests`, much like `cordova` is an alias to `cordova/node` in the current version.

---

> I am not against starting to use scope packages.

Me neither, we just have to decide something. As I said, the nested config names get a bit uglier. Then again, scoped packages _should_ be the future of Cordova. So, is the future now? :wink: 

---

> Basically what this might mean is creating two package, one being our core plugin called `cordova-eslint` which has the dependencies set in `package.json` as `dependencies` and is installed on each repo, and also the core package would also contain our new `eslint-config-cordova` or plugin.

:+1: This would definitely be the cleaner approach. Of course we would have to do two releases, each time we want to make an update. But if we would be using release candidate versions (like `2.0.0-rc.1`) then we could do the releases simultaneously IMHO.

**Comment by erisu** [comment]
> As I said, cordova/tests would also only be an alias to cordova/node-tests, much like cordova is an alias to cordova/node in the current version.

Then I am OK with `cordova/tests`

> So, is the future now? 😉

Yeah future is now I would say. So lets do scoped packages. Its only dev related stuff at this moment so it's a good test.

> release candidate

Will need future discussion outside of this since there might be more parts to this topic thats missing.

> then we could do the releases simultaneously IMHO.

Might be challenging since the `package.json` of one will need the new version of the other. So might be a 4 day release process. But, config might not need updating too often since its just configs and the dependencies would be peer. Should be met for the most part.
